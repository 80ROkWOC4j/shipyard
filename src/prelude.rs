pub use crate::get::GetComponent;
pub use crate::iter::{iterators, IntoIter, Shiperator};
pub use crate::not::Not;
pub use crate::remove::Remove;
pub use crate::run::System;
#[doc(hidden)]
pub use crate::run::SystemData;
pub use crate::sparse_set::{sort, sort::Sortable, View, ViewMut};
pub use crate::storage::{AllStorages, Entities, EntitiesMut, Key};
pub use crate::world::World;
pub use crate::Unique;
#[doc(hidden)]
#[cfg(feature = "proc")]
pub use shipyard_proc::system;

#[cfg(feature = "parallel")]
pub use crate::ThreadPool;
