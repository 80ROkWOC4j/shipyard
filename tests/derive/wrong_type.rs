use shipyard::prelude::*;

#[system(Test)]
fn run(_: fn()) {}

fn main() {}
