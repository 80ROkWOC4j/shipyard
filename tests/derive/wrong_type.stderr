error: A system will only accept a type of this list:
        - &T for an immutable reference to T storage
        - &mut T for a mutable reference to T storage
        - &Entities for an immutable reference to the entity storage
        - &mut EntitiesMut for a mutable reference to the entity storage
        - &mut AllStorages for a mutable reference to the storage of all components
        - &ThreadPool for an immutable reference to the rayon::ThreadPool used by the World
 --> $DIR/wrong_type.rs:4:11
  |
4 | fn run(_: fn()) {}
  |           ^^^^
